sdk/core/Azure.Core/src/Shared/PemReader.cs

```diff
@@ -146,13 +146,17 @@ private static X509Certificate2 CreateRsaCertificate(byte[] cer, byte[] key, X50
                 }
 
                 using X509Certificate2 certificateWithoutPrivateKey = new X509Certificate2(cer, (string)null, keyStorageFlags);
-                X509Certificate2 certificate = (X509Certificate2)s_rsaCopyWithPrivateKeyMethod.Invoke(null, new object[] { certificateWithoutPrivateKey, privateKey });
 
+                X509Certificate2 certificate = (X509Certificate2)s_rsaCopyWithPrivateKeyMethod.Invoke(null, new object[] { certificateWithoutPrivateKey, privateKey });
                 // On .NET Framework the PrivateKey member is not initialized after calling CopyWithPrivateKey.
+
+                // This class only compiles against NET6.0 in tests and never in SDK libraries suppress the warning
```

Comment:
This is not true. It ships with the Key Vault SDK and, IIRC, Azure.Identity.

Reply:
But in those libraries, it's compiled against netstandard2.0. None of our SDKs target net6.0 yet.